test/sys/test_sockopt.rs

```diff
@@ -281,6 +281,28 @@ fn test_tcp_congestion() {
     assert_eq!(getsockopt(&fd, sockopt::TcpCongestion).unwrap(), val);
 }
 
+#[test]
+#[cfg(target_os = "freebsd")]
+fn test_tcp_function_blk() {
+    use std::ffi::CStr;
+
+    let fd = socket(
+        AddressFamily::Inet,
+        SockType::Stream,
+        SockFlag::empty(),
+        None,
+    )
+    .unwrap();
+
+    let tfs = getsockopt(&fd, sockopt::TcpFunctionBlk).unwrap();
+    let name = unsafe{ CStr::from_ptr(tfs.function_set_name.as_ptr()) };
+    assert!(!name.is_empty());
```

Comment:
This function has not been stabilized in our MSRV 1.69, we can use:

```suggestion
    assert!(!name.to_bytes().is_empty());
```